test/FunctionalTests/Microsoft.OData.Core.Tests/Json/ODataUtf8JsonWriterTests.cs

```diff
@@ -0,0 +1,453 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriterTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Encodings.Web;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Json;
+using Xunit;
+
+namespace Microsoft.OData.Tests.Json
+{
+    /// <summary>
+    /// Unit tests for the ODataUtf8JsonWriter class
+    /// </summary>
+    public sealed class ODataUtf8JsonWriterTests: IDisposable
+    {
+        private IJsonWriter writer;
+        private MemoryStream stream;
+        private bool disposed;
+
+        public ODataUtf8JsonWriterTests()
+        {
+            this.stream = new MemoryStream();
+
+            try
+            {
+                this.writer = new ODataUtf8JsonWriter(stream, isIeee754Compatible: true, encoding: Encoding.UTF8, leaveStreamOpen: true);
+            }
+            catch

```

Comment:
Do we know which exceptions may be thrown when the `ODataUtf8JsonWriter` class is initialized? How are those exceptions handled from `ODataJsonLightOutputContext`? Should `ODataJsonLightOutputContext` the exception(s) and default back to `JsonWriter`?